### PR TITLE
These are the VFD conf files from https://github.com/ophub/amlogic-s9…

### PR DESCRIPTION
**Pull request by phodina**
Hi,

I found these configuration files for VFD driver/daemon on unofficial repository for Armbian.

The configuration for X96 Max Plus works for me and it would be great to merge them into one.

With the permission of the author.

### CODE REVIEW

File: hk1-x3.conf
Comment (arthur-liberman) [inline hk1-x3.conf:1]: This config already exists.

File: hk1box.conf
Comment (arthur-liberman) [inline hk1box.conf:1]: This looks identical to h96max-x3, so maybe redundant?

File: t95.conf
Comment (arthur-liberman) [inline t95.conf:1]: Does this [t95m-s905x-vfd](https://github.com/arthur-liberman/vfd-configurations/blob/master/t95m-s905x-vfd.conf) have the wrong config? Or are there multiple S905X variations? I'm pretty sure I still have a S905X T95m, and the config in my repo definitely works.
If this config is correct, please rename it to `t95m-s905x-2-vfd.conf` and I will rename the existing one with `1`.

File: t95z-plus.conf
Comment (arthur-liberman) [inline t95z-plus.conf:1]: Dup of [t95z-1-vfd.conf](https://github.com/arthur-liberman/vfd-configurations/blob/master/t95z-1-vfd.conf)

File: tx3-mini.conf
Comment (arthur-liberman) [inline tx3-mini.conf:1]: Please rename to `tanix-tx3-mini-vfd.conf`

File: tx3.conf
Comment (arthur-liberman) [inline tx3.conf:1]: Same as above

File: tx9-pro.conf
Comment (arthur-liberman) [inline tx9-pro.conf:1]: Dup [tanix-tx9-pro-vfd.conf](https://github.com/arthur-liberman/vfd-configurations/blob/master/tanix-tx9-pro-vfd.conf)

File: whale.conf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Share by: https://github.com/9runner
+#--------------------
+#gpio_xxx:
+# [0] 0 = &gpio, 1 = &gpio_ao.
+# [1] pin number - https://github.com/openSUSE/kernel/blob/master/include/dt-bindings/gpio/meson-g12a-gpio.h
+# [0] Reserved - must be 0.
+
+vfd_gpio_clk='0,64,0'
+vfd_gpio_dat='0,63,0'
+vfd_gpio_stb='1,10,0'
```
Comment (arthur-liberman) [inline whale.conf:13]: Considering the controller here is the FD6551, which has a 2-wire interface, I would say that this line is probably incorrect and should be ` vfd_gpio_stb='0,0,0xFF'`

File: x92.conf
Comment (arthur-liberman) [inline x92.conf:1]: Dup [x92-vfd.conf](https://github.com/arthur-liberman/vfd-configurations/blob/master/x92-vfd.conf)

File: x96max.conf
Comment (arthur-liberman) [inline x96max.conf:1]: Same as x96air, maybe worth unifying them?

File: x96maxplus.conf
Comment (arthur-liberman) [inline x96maxplus.conf:1]: Same as above.

**Review by arthur-liberman — CHANGES_REQUESTED** [review]
Can you please rename the files to follow the naming convention that the rest of the vfd files have?

**Comment by arthur-liberman** [comment]
Thanks for this, but I made some comments, as I can't accept it as-is.

File: hk1-x3.conf
Comment (phodina) [reply hk1-x3.conf:1]: Removed

File: hk1box.conf
Comment (phodina) [reply hk1box.conf:1]:
```
--- hk1box.conf	2023-06-10 00:27:36.961211989 +0200
+++ h96max-x3.conf	2023-06-10 00:27:36.961211989 +0200
@@ -1,6 +1,6 @@
 # This file must be renamed to vfd.conf and placed in the /storage/.config/ folder.
 #
-# HK1Box / Vontar X3 (S905X3) configuration
+# H96 Max X3 (S905X3) configuration
 #--------------------
 #gpio_xxx:
 # [0] 0 = &gpio, 1 = &gpio_ao.
@@ -35,4 +35,4 @@

 vfd_display_type='0x00,0x00,0x00,0x06'

-functions='alarm usb play pause colon lan wlan'
+functions='alarm usb play pause colon eth wifi'
```

Agreed!

File: t95z-plus.conf
Comment (phodina) [reply t95z-plus.conf:1]: Removed.

File: tx3-mini.conf
Comment (phodina) [reply tx3-mini.conf:1]: Done!

File: tx9-pro.conf
Comment (phodina) [reply tx9-pro.conf:1]:
```
--- tx9-pro.conf	2023-06-10 00:27:36.961211989 +0200
+++ tanix-tx9-pro-vfd.conf	2023-06-10 00:27:17.505131843 +0200
@@ -1,14 +1,14 @@
 # This file must be renamed to vfd.conf and placed in the /storage/.config/ folder.
 #
-# Tanix TX9 Pro (S912) configuration
+# Tanix TX9 Pro configuration
 #--------------------
 #gpio_xxx:
 # [0] 0 = &gpio, 1 = &gpio_ao.
 # [1] pin number - https://github.com/openSUSE/kernel/blob/master/include/dt-bindings/gpio/meson-gxl-gpio.h
 # [0] Reserved - must be 0.

-vfd_gpio_clk='0,75,0'
-vfd_gpio_dat='0,74,0'
+vfd_gpio_clk='0,76,0'
+vfd_gpio_dat='0,75,0'
 vfd_gpio_stb='0,53,0'

 #chars:
@@ -34,6 +34,3 @@
 # [3] - Controller.

 vfd_display_type='0x00,0x00,0x00,0x00'
-
-#functions='alarm usb play pause colon lan wlan'
-functions=''
```

Are you sure? The `vfd_gpio_clk` and `vfd_gpio_dat` are allocated to different pins.

File: x92.conf
Comment (phodina) [reply x92.conf:1]: They are same. So removing

File: whale.conf (hunk 1 above)
Comment (phodina) [reply whale.conf:13]: Unfortunately I don't own the device and don't have that much experience with this chip. So I modified the config as suggested.

Can be changed back if somebody discovers it does not work.

File: t95.conf
Comment (phodina) [reply t95.conf:1]: I can't guarantee the validity of this config as I don't own the device. If you want I can just remove this device and use your configuration.

File: tx3.conf
Comment (phodina) [reply tx3.conf:1]: Done

**Comment by phodina** [comment]
Hi @arthur-liberman,

thanks for the review. I've applied most of your points.

File: h96max-x3-vfd.conf
Comment (arthur-liberman) [inline h96max-x3-vfd.conf:1]: h96-max-x3-vfd.conf

File: hk1box-vfd.conf
Comment (arthur-liberman) [inline hk1box-vfd.conf:1]: hk1-box-vfd.conf

File: t95-vfd.conf
Comment (arthur-liberman) [inline t95-vfd.conf:1]: t95m-s905x-2-vfd.conf

File: tanix-tx3-vfd.conf
Comment (arthur-liberman) [inline tanix-tx3-vfd.conf:1]: There was an older Tanix TX3 box, you overwrote its config.
You can add `tanix-tx3-s905x3-vfd.conf` or add `-2-` before `vfd.conf`

File: tx9-pro.conf
Comment (arthur-liberman) [inline tx9-pro.conf:1]: Is it a Tanix box?
I think so, it should be `tanix-tx9-pro-2-vfd.conf` and the existing tx9 pro file may be changed to `tanix-tx9-pro-1-vfd.conf`

**Review by arthur-liberman — COMMENTED** [review]
Thanks!
Mostly nitpicking, but I'd like names to be consistent. It makes it easier for me to track/find things later.

**Comment by phodina** [comment]
No problem. I also like consistency at least in names since the component-wise it's a mess on these boxes :-D

**Review by arthur-liberman — APPROVED** [review]

**Comment by arthur-liberman** [comment]
This is great, thank you.

**Pull request merged**